src/main/java/com/yegor256/tojos/MnSynchronized.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2016-2022 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.yegor256.tojos;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * This class is thread-safe.
+ *
+ * @since 0.3.0
+ */
+public final class MnSynchronized implements Mono {
+
+    /**
+     * The wrapped mono.
+     */
+    private final Mono wrapped;
+
+    /**
+     * Ctor.
+     *
+     * @param mono The mono
+     */
+    public MnSynchronized(final Mono mono) {
+        this.wrapped = mono;
+    }
+
+    @Override
+    public Collection<Map<String, String>> read() {
+        synchronized (this.wrapped) {
+            return Collections.synchronizedCollection(this.wrapped.read());
```

Comment:
@l3r8yJ I think `Collections.synchronizedCollection` is redundant here

Reply:
@yegor256 i'm think i wrote the wrong test or `Collections.synchronizedCollection` isn't redurant
```
java.lang.AssertionError: 
Expected: <1000>
     but: was <0>
Expected :<1000>
Actual   :<0>
```

Reply:
@yegor256 yes, there is some mess with the test , I got a little confused about `writing` and `reading` **all** the data **simultaneously** from the source. In this test, the last thread always rewrites `rows` with a `size` equal to the number of threads